sdks/jreleaser-git-java-sdk/src/main/java/org/jreleaser/sdk/git/ChangelogGenerator.java

```diff
@@ -597,30 +617,135 @@ private void addContributor(String name, String email) {
         }
 
         static Commit of(RevCommit rc) {
-            Commit c = new Commit();
-            c.fullHash = rc.getId().name();
-            c.shortHash = rc.getId().abbreviate(7).name();
-            c.body = rc.getFullMessage();
-            String[] lines = split(c.body);
-            c.title = lines[0];
-            c.author = new Author(rc.getAuthorIdent().getName(), rc.getAuthorIdent().getEmailAddress());
-            c.addContributor(rc.getCommitterIdent().getName(), rc.getCommitterIdent().getEmailAddress());
-            c.time = rc.getCommitTime();
-            for (String line : lines) {
-                Matcher m = CO_AUTHORED_BY_PATTERN.matcher(line);
-                if (m.matches()) {
-                    c.addContributor(m.group(1), m.group(2));
-                }
-            }
-            return c;
+            return new Commit(rc);
         }
 
-        private static String[] split(String str) {
+        protected static String[] split(String str) {
             // Any Unicode linebreak sequence
             return str.split("\\R");
         }
     }
 
+    static class ConventionalCommit extends Commit {
+        private static final Pattern FIRST_LINE_PATTERN =
+            Pattern.compile("^(?<type>build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(?:\\((?<scope>\\w+)\\))?(?<bang>!)?: (?<description>.*$)");
```

Comment:
Previously they were handled in the replacer regex and the categories are also still there. We could extract the scopes in configuration, or make the regex generic to not check each commit type but accept any word instead.

Second option seems more flexible. WDYT?

Reply:
Note that deps is usually passed as scope of a build commit.

Reply:
That is right, but previously scopes were defined as a set of labelers which could be overridden or extended with additional entries. 2nd option (word regex) sounds better, as JReleaser itself uses `i18n` and `deps` as additional scopes.

Reply:
FWIW all Kordamp projects & JReleaser use `deps` as scope instead of `build`.

Reply:
The labelers are still used here for the categories, only the parsing would fail in that case, and the commit would be considered as a normal commit. Will change this to word regex all lowercase.